s3storageprovider/src/main/java/org/duracloud/s3storage/S3StorageProvider.java

```diff
@@ -1020,4 +1033,39 @@ protected String getWithSpace(String name) {
         return name.replaceAll("%20", " ");
     }
 
+    static protected String encodeHeaderValue(String userMetaValue) {
```

Comment:
It would be helpful to add a comment here or somewhere else relevant that points to https://tools.ietf.org/html/rfc5987#section-3.2.2 and indicates that encoding of header values is being handled in compliance with this spec.